components/raftstore/src/store/peer_storage.rs

```diff
@@ -1002,23 +1026,23 @@ where
             return Ok(prev_last_index);
         }
 
+        invoke_ctx.has_new_entries = true;
+
         let (last_index, last_term) = {
             let e = entries.last().unwrap();
             (e.get_index(), e.get_term())
         };
 
-        for entry in entries {
-            if !ready_ctx.sync_log() {
-                ready_ctx.set_sync_log(get_sync_log_from_entry(entry));
-            }
-        }
+        let update_cache = if let Some(ref mut cache) = self.cache {
+            Some((cache, entries.clone()))
+        } else {
+            None
+        };
```

Comment:
Why we need `update_cache`? Also, maybe we can append entries to the cache first then the `WriteBatch` to avoid cloning.

Reply:
You are right. The code is updated.